### ISSUE TITLE
Broken macOS build from source

### ISSUE THREAD

**Issue by caco-underscore** [xmoto/xmoto#193]
When building from source on macOS for arm64 arch, it opens and immediately closes. Launching it from console, it says:
```
fatal exception : invalid or missing texture file (Textures/UI/Misc.png)
```
This is the verbose and debug output:
```
LANGUAGE=
Wanted resolution: 800x600
GL: Apple M1 Pro (Apple)
GL: using ARB_vertex_buffer_object
GL: using EXT_framebuffer_object
GL: using ARB_fragment_shader/ARB_vertex_shader/ARB_shader_objects
Resolution: 800x600
XmDb version is 37
Joystick mappings loaded
0 compatible controllers found
** Debug ** : Controller mappings:
Loading theme from file Themes/original.xml
** Warning ** : Edge with no scale
Using theme: Classic
** Warning ** : TextureManager::loadTexture() : texture 'Textures/UI/Misc.png' not found or invalid
UserUnload started at 0.266
net: number of TCP packets sent      : 0
net: biggest TCP packet sent         : 0 bytes
net: size of TCP packets sent        : 0 bytes
net: number of UDP packets sent      : 0
net: biggest UDP packet sent         : 0 bytes
net: size of UDP packets sent        : 0 bytes
net: number of TCP packets received  : 0
net: biggest TCP packet received     : 0 bytes
net: size of TCP packets received    : 0 bytes
net: number of UDP packets received  : 0
net: biggest UDP packet received     : 0 bytes
net: size of UDP packets received    : 0 bytes
** Debug ** : Open db for thread with key 'XMTHREADSTATS'
** Warning ** : sendAsynchronousMessage message [[STATS_UPDATED]s] sent and there's no state to receive it.
** Debug ** : UserUnload saveConfig at 0.351
** Debug ** : UserUnload saveConfig ended at 0.353
** Debug ** : ---Texture not freed automatically---
** Debug ** : --- --- ---
UserUnload ended at 0.357
fatal exception : invalid or missing texture file (Textures/UI/Misc.png)
```

**Comment by nikekson** [comment]
What's the command you used to build it? And did you bundle it as a `DragNDrop` bundle with CPack?

**Comment by caco-underscore** [comment]
I followed the instructions of the wiki for macOS:
```
git clone https://github.com/xmoto/xmoto
cd xmoto
mkdir build && cd build
cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_MACOS_BUNDLE=ON ..
make -j$(sysctl -n hw.ncpu)
cpack -G DragNDrop
```
When I open the app from the package 'xmoto-0.6.2-arm64.dmg', it shows the icon on the dock and quits immediately; launching it from console (like in ./X-Moto.app/Contents/MacOS/X-Moto) shows what it is in the first comment.

**Comment by caco-underscore** [comment]
Ok found the problem. At tim_png.cpp:106:
```
PngPtr = png_create_read_struct(
    PNG_LIBPNG_VER_STRING, (png_voidp)&Env, user_error_fn, user_warning_fn);
```
PngPtr is null, because PNG_LIBPNG_VER_STRING is different than the runtime libpng. For some unknown reason to me, cmake gets confused between macOS Mono and homebrew files, so it uses the libpng headers from macOS Mono but copies and link the homebrew lib into the bundle, which is a different version.
I have no experience with cmake, so I just copied a solution from stackoverflow into src/CMakeLists.txt, replacing ```
find_package(PNG REQUIRED)```[https://stackoverflow.com/questions/30980383/cmake-compile-options-for-libpng](url):
```
# search for pkg-config
include (FindPkgConfig)
if (NOT PKG_CONFIG_FOUND)
    message (FATAL_ERROR "pkg-config not found")
endif ()

# check for libpng
pkg_check_modules (LIBPNG libpng16 REQUIRED)
if (NOT LIBPNG_FOUND)
    message(FATAL_ERROR "You don't seem to have libpng16 development libraries installed")
else ()
    include_directories (${LIBPNG_INCLUDE_DIRS})
    link_directories (${LIBPNG_LIBRARY_DIRS})
    link_libraries (${LIBPNG_LIBRARIES})
endif ()
add_executable (app_png ${_MYSOURCES} ${LIBPNG_LINK_FLAGS})
```
and it worked after I deleted the last line (cmake complained). Now the game launches and plays fine. I guess someone that knows cmake can come with a proper solution.

**Comment by nikekson** [comment]
Fixed for real by #195
